### ISSUE TITLE
AP6330 not working

### ISSUE THREAD

**Issue by DiogoSilva48** [dtechsrv/LibreELEC.tv#2]
Hi, I can't get my wifi/bluetooth working. Using this version https://forum.libreelec.tv/thread/2877-s802-s812-libreelec-images-for-7-8-9/ I get my AP6330 device using the MXIII Plus dtb. I tried switching he dtb's and multiple builds or your libreelec and nothing changed. Any tips or anything I can do to help fix ?

**Comment by dtechsrv** [comment]
Which device did you try my firmware image? Please describe the specific vendor and model number.

**Comment by DiogoSilva48** [comment]
The problem is I don't know exactly which device I'm running, I know it's a S812 from Geniatech / Mygica, which has a AP6330, and the "list of devices" which includes mine are called stvm8. Weirdly enough my device seems to run perfectly fine everything ROM related to MX III Plus and MX III - G and a couple of M8SS ROM's which support AP6330. Weirdest thing for sure.

**Comment by dtechsrv** [comment]
Can you send me a dmesg log?

**Comment by DiogoSilva48** [comment]
> Can you send me a dmesg log?

Yes, for sure. I'll send you tomorrow. Which build should I do it on ?

**Comment by dtechsrv** [comment]
It doesn't really matter which build you choose.

**Comment by DiogoSilva48** [comment]
Hi. I am trying to get ethernet working and although it shows connected it's not working at all, even tried to configure ipv4 manually but no luck. Any other way to get the dmesg other than ssh ?

**Comment by dtechsrv** [comment]
This usually happens when the device is crossflashed with an incompatible firmware, because it is overwrite the factory bootloader.

Do you still have the factory ROM on your device?

**Comment by DiogoSilva48** [comment]
I don't have my factory ROM on my device because I wasn't able to boot LibreElec from my stock firmware before but I can try again, maybe replacing just the recovery will do the trick, I never tried before. I'll let you know. Thanks.

**Comment by dtechsrv** [comment]
Basically, recovery is not important, much more the uboot bootloader, because the bus addresses of the integrated devices on the PCB are hardwired there.

**Comment by DiogoSilva48** [comment]
> Basically, recovery is not important, much more the uboot bootloader, because the bus addresses of the integrated devices on the PCB are hardwired there.

I am going to try to get LibreElec booted on my stock firmware and let you know, thanks 😅. The problem I had I thought it was actually recovery because I was getting errors with the amlautoscript flashing, some verification.

**Comment by DiogoSilva48** [comment]
I was finally able to get ethernet working, found a firmware with a heck of a support, on mxiii plus build now I have ethernet + wifi, on yours just ethernet, anyway I'm attaching the dmesg, sorry for the delay, been busy just found time today to do some research.
[dmesg.txt](https://github.com/dtechsrv/LibreELEC.tv/files/5484844/dmesg.txt)

**Comment by dtechsrv** [comment]
I'm glad it was solved. Basically, the MXIII Plus and the M8S+ are quite a lot of things different, if you just look at the dtb. Therefore, it probably won't start without the correct DTB.

I checked the log you sent and it seems that the dtb is causing the problem. In essence, I don’t want to touch my code base, because I only have the tools listed on the main page and can only test on them.

Regardless, thank you. ;)
